airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py

```diff
@@ -275,6 +284,27 @@ def _download_report(self, report_info: ReportInfo, url: str) -> List[dict]:
         """
         return super()._download_report(None, url)
 
+    @backoff.on_exception(
+        backoff.expo,
+        (
+            requests.exceptions.Timeout,
+            requests.exceptions.ConnectionError,
+            TooManyRequests,
+        ),
+        max_tries=10,
+    )
+    def _send_http_request(self, url: str, profile_id: int, json: dict = None):
+        headers = self._get_auth_headers(profile_id)
+        if json:
```

Comment:
I don't know a lot about source-amazon-ads but it feels like there could be requests where we don't have `json` but we still want the authentication. For example with other sources [like salesforce](https://github.com/airbytehq/airbyte/blob/6a63ffa1893e4d1a27c7b4b3610b303403ebcdc5/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L406), getting the job status requires the authentication but does not have a body. It feels like this is also the case for this source where the `json` parameter is not passed: https://github.com/airbytehq/airbyte/blob/6a63ffa1893e4d1a27c7b4b3610b303403ebcdc5/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L274

Reply:
yes, you are right for check status logic requires authentication via headers.  `_get_auth_headers` will return headers in case `profile_id` were provided, otherwise it's empty: [code line](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L249).

So it makes us think that no auth headers will be presented in request when we were using `self._session`. But auth attribute was set  for `self._session` [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L123) and requests lib merges empty header with auth headers from auth atr in `self._session` if it wasn't added to the headers. That is the  reason why we had 400 status code for download for our test account and for airbyte users. Wrong request contained both Auth header and X-Amz-Algorithm query parameter, Signature query string parameters.

Why check status works with _no_auth_session: when the source checks status for the report uses profile_id, and in case of provided if `_get_auth_headers` [returns](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L249) auth headers from `self._session`, so the request is properly authenticated.

Why download works with  _no_auth_session: we don't need profile id to download report, so _get_auth_headers == {} and requests lib don't merge auth headers inside of it, because auth atr is None. The request made properly, without `Only one auth mechanism allowed; only the X-Amz-Algorithm query parameter, Signature query string parameter or the Authorization header should be specified` error, all needed query param for auth already present in the url.

Reply:
Ok! So when we do [this HTTP request](https://github.com/airbytehq/airbyte/blob/2f85ba76fd7f51a5a5d2bd93ce70783086bc87db/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L279), we don't need the authentication?

Reply:
[this HTTP request](https://github.com/airbytehq/airbyte/blob/2f85ba76fd7f51a5a5d2bd93ce70783086bc87db/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L279): auth is required for check status

Reply:
If the parameter `json` is not explicit in this method call though, it feels like we will fall in the "we don't use auth"  branch in [this method](https://github.com/airbytehq/airbyte/blob/2f85ba76fd7f51a5a5d2bd93ce70783086bc87db/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py#L296-L306)

Like, reading this, it feels like the check status will not provide authentication

Reply:
Yep, I also find this confusing, I will use your suggesting from this [comment](https://github.com/airbytehq/airbyte/pull/43354#discussion_r1709441307) to make it more understandable.

Reply:
updated sessions